core/services/vrf/listener_v2.go

```diff
@@ -990,6 +990,19 @@ func (lsn *listenerV2) simulateFulfillment(
 	res.run, trrs, err = lsn.pipelineRunner.ExecuteRun(ctx, *lsn.job.PipelineSpec, vars, lg)
 	if err != nil {
 		res.err = errors.Wrap(err, "executing run")
+
+		// For an errored pipeline run, wait until the finality depth of the chain to be cleared,
+		// then check if the failing request is being called by an invalid sender. If this is the case,
+		// drop the request.
+		latestHead := lsn.getLatestHead()
+		if latestHead-req.req.Raw.BlockNumber > uint64(lsn.cfg.EvmFinalityDepth()) {
+			code, err := lsn.ethClient.CodeAt(ctx, req.req.Sender, big.NewInt(int64(latestHead)))
+			if err != nil || len(code) == 0 {
```

Comment:
Should we log error message when `err != nil`? Just in case RPC call constantly fails for whatever reason

Reply:
Yeah true I should handle the error a bit better.

Reply:
Force-pushed so comment context lost but should be resolved now.